File: packages/facebook-analytics/ios/Plugin/FacebookAnalyticsPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(FacebookAnalyticsPlugin, "FacebookAnalytics",
           CAP_PLUGIN_METHOD(initAppEvents, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(logEvent, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(enableAdvertiserTracking, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(disableAdvertiserTracking, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(getAdvertiserTrackingStatus, CAPPluginReturnPromise);
)

File: packages/facebook-analytics/ios/Plugin/FacebookAnalyticsPlugin.swift
import Foundation
import Capacitor
import FBSDKCoreKit

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(FacebookAnalyticsPlugin)
public class FacebookAnalyticsPlugin: CAPPlugin {

    @objc func initAppEvents(_ call: CAPPluginCall) {
        AppEvents.shared.activateApp()
        call.resolve()
    }

    @objc func enableAdvertiserTracking(_ call: CAPPluginCall) {
        Settings.shared.isAdvertiserTrackingEnabled = true
        call.resolve()
    }

    @objc func disableAdvertiserTracking(_ call: CAPPluginCall) {
        Settings.shared.isAdvertiserTrackingEnabled = false
        call.resolve()
    }

    @objc func getAdvertiserTrackingStatus(_ call: CAPPluginCall) {
        print("facebook advertiser tracking status: ", Settings.shared.isAdvertiserTrackingEnabled)
        call.resolve(["status": Settings.shared.isAdvertiserTrackingEnabled])
    }

    @objc func logEvent(_ call: CAPPluginCall) {
        guard let event = call.getString("event") else {
            call.reject("Missing event argument")
            return;
        }

        print("sending facebook event: ", event)

        if let valueToSum = call.getDouble("valueToSum") {
            if let params = call.getString("params") {
                let newParams = [
                    AppEvents.ParameterName.content : params,
                ]
                AppEvents.shared.logEvent(.init(event), valueToSum: valueToSum, parameters: newParams)
            } else {
                AppEvents.shared.logEvent(.init(event), valueToSum: valueToSum)
            }
        } else {
            if let params = call.getString("params") {
                print(params)
                let newParams = [
                    AppEvents.ParameterName.content : params,
                ]
                AppEvents.shared.logEvent(.init(event), parameters: newParams)
            } else {
                AppEvents.shared.logEvent(.init(event))
            }
        }
        call.resolve()
    }

}
